flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -524,7 +524,7 @@ extractExecutionResults() {
 
     execution_results=$(echo "${output}" | grep ${EXECUTION_PREFIX})
     num_lines=$(echo "${execution_results}" | wc -l)
-    if [[ ${num_lines} -ne ${expected_lines} ]]; then
+    if [[ -z ${execution_results} || ${num_lines} -ne ${expected_lines} ]]; then
```

Comment:
I would suggest to handle this case separately to output a better error message (e.g. "The execution result is empty.").

Reply:
yes, the current error message would be misleading since it would claim that there was 1 line.